src/main/kotlin/net/lumalyte/lg/LumaGuilds.kt

```
@@ -979,6 +986,35 @@ class LumaGuilds : JavaPlugin() {
+        try {
+            val channelManager = dev.rosewood.rosechat.api.RoseChatAPI.getInstance().channelManager
+            channelManager.register(LumaGuildsChannelProvider())
+
+            // One-tick delay: RoseChat has already tried to load channels at
+            // startup and found no registered LumaGuilds provider. Now that the
+            // provider exists, re-read channels.yml.
+            Bukkit.getScheduler().runTaskLater(this, Runnable {
+                try {
+                    channelManager.reload()
+                    logColored("✓ LumaGuilds RoseChat channels registered (guild, guild-ally, guild-modchat)")
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)